soc/x86/apollo_lake/Kconfig.soc

```diff
@@ -5,7 +5,7 @@ config SOC_APOLLO_LAKE
 	bool "Intel Apollo Lake Soc"
 	select X86
 	select CPU_APOLLO_LAKE
-	select HAS_I2C_DW
+	select HAS_I2C_DW if I2C
```

Comment:
hmm, I think there is a more fundamental issue here, i.e. HAS_I2C_DW should not depend on I2C at all. HAS_I2C_DW provides a capability and does not really enable anything and should not depend on anything. But for now this is good enough I guess and I see this done for S1000 as well.

Reply:
Agreed. @jhedberg do you want to provide a more thourougj solution?

Reply:
@carlescufi i do, but because of its invasiveness I’d leave it after the release, ie merge this one first